### PR TITLE
[FLINK-26342][docs][formats] Add DataStream documentation for the CSV format

### PR DESCRIPTION
**Pull request by afedulov**
## What is the purpose of the change

This PR adds documentation for the DataStream CSV format.

## Verifying this change

This change is a trivial addition to the documentation without any test coverage.

### CODE REVIEW

File: docs/content/docs/connectors/datastream/formats/csv.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+CsvReaderFormat<T> forSchema(CsvMapper mapper, 
+                             CsvSchema schema, 
+                             TypeInformation<T> typeInformation) 
+```
+
+Similarly to the `TextLineInputFormat`, `CsvReaderFormat` can be used in both continues and batch modes (see [TextLineInputFormat]({{< ref "docs/connectors/formats/text_files" >}}  for examples).
+
+### Write
+
+For writing CSV files, use `CsvBulkWriter`. An example of producing CSV files from a stream of POJOs  looks like this:
```
Comment (fapaul) [inline docs/content/docs/connectors/datastream/formats/csv.md:66]:
```suggestion
For writing CSV files, use `CsvBulkWriter`. An example of producing CSV files from a stream of POJOs looks like this:
```

File: docs/content/docs/connectors/datastream/formats/csv.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+```
+
+Similarly to the reader, the writer also utilizes Jackson library and supports fully customizable configuration using its `CsvMapper` and `CsvSchema` classes:
+
+```java
+
+Converter<CityPojo, CityPojo, Void> converter = (value, context) -> value; //An optional custom types converter, no-op in this example 
+CsvMapper csvMapper = new CsvMapper();
+CsvSchema schema = csvMapper.schemaFor(CityPojo.class).withColumnSeparator("|");;
+BulkWriter.Factory<CityPojo> bulkWriterFactory = (out) -> CsvBulkWriter.forSchema(csvMapper, schema, converter, null, out);
```
Comment (fapaul) [inline docs/content/docs/connectors/datastream/formats/csv.md:86]: Does this really work? The class `CsvBulkWriter` seems to be package-private. I suppose we are not supporting writing Csv files outside of the Table API?

**Review by fapaul — CHANGES_REQUESTED** [review]
Content looks pretty good but can you check the failing CI. I think the docs won't build because of a wrong reference.

I am also unsure about the writing part it feels rather hard to use from the DataStream API in the current state.

File: docs/content/docs/connectors/datastream/formats/csv.md (hunk 2 above)
Comment (afedulov) [reply docs/content/docs/connectors/datastream/formats/csv.md:86]: Hmm, indeed. I think Arvid proposed to keep it so, but that was before we added convenience static factory methods. It was not really defined as a requirement, but I do not see why not expose it in the next version.

Comment (afedulov) [reply docs/content/docs/connectors/datastream/formats/csv.md:86]: Hmm, indeed. I think Arvid proposed to keep it so, but that was before we added convenience static factory methods. It was not really defined as a requirement, but I do not see why not expose it in the next version.

Comment (afedulov) [reply docs/content/docs/connectors/datastream/formats/csv.md:86]: Hmm, indeed. I think Arvid proposed to keep it so, but that was before we added convenience static factory methods. It was not really defined as a requirement, but I do not see why not expose it in the next version.

File: docs/content/docs/connectors/datastream/formats/csv.md (hunk 1 above)
Comment (afedulov) [reply docs/content/docs/connectors/datastream/formats/csv.md:66]: Removed this part.

**Comment by afedulov** [comment]
Thanks for the review, @fapaul. I fixed the build and removed the writer part as it is indeed only currently meaningful in the SQL context.

**Review by fapaul — APPROVED** [review]
Thanks for addressing the comment. Can you squash the commits together?

**Comment by afedulov** [comment]
Thanks, @fapaul. Done.

**Comment by fapaul** [comment]
Docs CI has passed I am merging

**Pull request merged**